projects/plugins/crm/src/automation/commons/conditions/contacts/class-contact-tag.php

```diff
@@ -0,0 +1,209 @@
+			$this->condition_met = false;
+
+			return;
+		}
+
+		if ( ! $this->is_valid_contact_tag_data( $data ) || ! $this->is_valid_contact_tag_data( $previous_data ) ) {
+			$this->logger->log( 'Invalid contact tag data' );
+			$this->condition_met = false;
+
+			return;
+		}
+
+		$operator = $this->get_attributes()['operator'];
+		$tag      = $this->get_attributes()['tag'];
+
+		$this->check_for_valid_operator( $operator );
+		$this->logger->log( 'Condition: Contact_Tag ' . $operator . ' => ' . $tag );
+
+		switch ( $operator ) {
+			case 'tag_added':
+				$this->condition_met = ( ! $this->has_tag_by_name( $previous_data, $tag ) && $this->has_tag_by_name( $data, $tag ) );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+			case 'tag_removed':
+				$this->condition_met = ( $this->has_tag_by_name( $previous_data, $tag ) && ! $this->has_tag_by_name( $data, $tag ) );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+			case 'has_tag':
+				$this->condition_met = $this->has_tag_by_name( $data, $tag );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+			default:
+				$this->condition_met = false;
+				throw new Automation_Exception(
+					/* Translators: %s is the unimplemented operator. */
+					sprintf( __( 'Valid but unimplemented operator: %s', 'zero-bs-crm' ), $operator ),
+					Automation_Exception::CONDITION_OPERATOR_NOT_IMPLEMENTED
+				);
+		}
+	}
+
+	/**
+	 * Checks if the contact has at least the necessary keys to detect a contact
+	 * tag condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param array $data The event data.
+	 * @return bool True if the data is valid to evaluate a contact tag condition, false otherwise.
+	 */
+	private function is_valid_contact_tag_data( array $data ): bool {
+		return is_array( $data ) && isset( $data['tags'] );
+	}
+
+	/**
+	 * Get the title for the contact tag condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return string The title.
+	 */
+	public static function get_title(): string {
+		return __( 'Contact Tag', 'zero-bs-crm' );
+	}
+
+	/**
+	 * Get the slug for the contact tag condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return string The slug 'jpcrm/condition/contact_tag'.
+	 */
+	public static function get_slug(): string {
+		return 'jpcrm/condition/contact_tag';
+	}
+
+	/**
+	 * Get the description for the contact tag condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return string The description for the condition.
+	 */
+	public static function get_description(): string {
+		return __( 'Checks if a contact tag matches a specified condition', 'zero-bs-crm' );
+	}
+
+	/**
+	 * Get the category of the contact tag condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return string The category 'contact'.
+	 */
+	public static function get_category(): string {
+		return 'contact';
```

Comment:
For now, we retrieved the label translated it, and capitalized it. Until we implement the Category class.
```suggestion
		return __( 'Contact', 'zero-bs-crm' );
```

Reply:
Addressed in 236c3ae